data_structures/src/chain.rs

```diff
@@ -635,6 +636,76 @@ impl SuperBlock {
         ]
         .concat()
     }
+
+    /// Generate a Proof of Inclusion for a given Superblock tally root, provided the DR transaction
+    /// and the blocks that the superblock contains
+    /// If the DR transaction is not found within those blocks, returns None
+    pub fn dr_proof_of_inclusion(
+        &self,
+        blocks: &[Block],
+        tx: DRTransaction,
+    ) -> Option<TxInclusionProof> {
+        // Get the PoI for the block root, if the data request is found on the list of blocks
+        // Obtain also the tally root of the block containing the Dr TX.
+        let (mut poi, current_root) = blocks.iter().find_map(|b| {
```

Comment:
You only use `current_root` to find the `dr_root_idx` later. You could get the index here directly:

```suggestion
        let (mut poi, dr_root_idx) = blocks.iter().enumerate().find_map(|(idx, b)| {
```

And return `Some((tx.data_proof_of_inclusion(b)?, idx))`